### DISCUSSION TITLE
Postgresql Advanced Search and tag search do not work in combination with Page Rules

### DISCUSSION THREAD

**Discussion by despens** [requarks/wiki#3664] [Error / Bug Report]
**Describe the bug**
When a wiki is set up with pages visible for the public (Guest users) and pages visible for logged-in users, Postgresql Advanced search doesn't work for Guest users and returns no results.

**To Reproduce**
1. Create a page and add a tag `public`.
2. Create a page rule for the user group `Guests`:
   `Allow` `read:pages` `Tag Matches…` `public`
   This will allow Guest users to only view pages that are tagged with `public`.
3. Confirm that not logged in users can access the newly created page, but not other pages.
4. As a logged-out user, perform a text search or tag search on the contents of the new page: the search returns no results.

**Expected behavior**
Search should present results based on what pages the current user has read access to.

**Host Info (please complete the following information):**
 - OS: Ubuntu 18.04, Docker
 - Wiki.js version: 2.2.50
 - Database engine: Postgresql 12.0

**Comment by despens** [comment]
Confirmed with version 2.3.81

**Reply by prognaest** [reply]
Still applies on version 2.5.295. Migitatable by adding a page rule to allow read access to all sites or switching to another provider
